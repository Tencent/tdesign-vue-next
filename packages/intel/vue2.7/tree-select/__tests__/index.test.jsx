import { mount } from '@vue/test-utils';
import TreeSelect from '@/src/tree-select/index.ts';

const options = [
  {
    label: '广东省',
    value: 'guangdong',
    children: [
      {
        label: '广州市',
        value: 'guangzhou',
      },
      {
        label: '深圳市',
        value: 'shenzhen',
      },
    ],
  },
  {
    label: '江苏省',
    value: 'jiangsu',
    children: [
      {
        label: '南京市',
        value: 'nanjing',
      },
      {
        label: '苏州市',
        value: 'suzhou',
      },
    ],
  },
];

const propsOptions = [
  {
    name: '广东省',
    pinyin: 'guangdong',
    children: [
      {
        name: '广州市',
        pinyin: 'guangzhou',
      },
      {
        name: '深圳市',
        pinyin: 'shenzhen',
      },
    ],
  },
  {
    name: '江苏省',
    pinyin: 'jiangsu',
    children: [
      {
        name: '南京市',
        pinyin: 'nanjing',
      },
      {
        name: '苏州市',
        pinyin: 'suzhou',
      },
    ],
  },
];

const objectValue = { label: '深圳市', value: 'shenzhen' };

// every component needs four parts: props/events/slots/functions.
describe('TreeSelect', () => {
  // test props api
  describe(':props', () => {
    it(':clearable', () => {
      const wrapper = mount({
        render() {
          return <TreeSelect clearable={true} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':disabled', () => {
      const wrapper = mount({
        render() {
          return <TreeSelect disabled={true} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':readonly', () => {
      const wrapper = mount({
        render() {
          return <TreeSelect disabled={true} />;
        },
      });
      expect(wrapper.find('.t-is-disabled').exists()).toBe(true);
    });
    it(':empty string', () => {
      const wrapper = mount({
        render() {
          return <TreeSelect empty={'no data'} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':filterable', () => {
      const wrapper = mount({
        render() {
          return <TreeSelect filterable={true} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':loading', () => {
      const wrapper = mount({
        render() {
          return <TreeSelect loading={true} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':loadingText', () => {
      const wrapper = mount({
        render() {
          return <TreeSelect loadingText={'data loading'} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':max', () => {
      const wrapper = mount({
        render() {
          return <TreeSelect max={10} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':minCollapsedNum', () => {
      const wrapper = mount({
        render() {
          return <TreeSelect minCollapsedNum={1} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':multiple', () => {
      const wrapper = mount({
        render() {
          return <TreeSelect multiple={true} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':placeholder', () => {
      const wrapper = mount({
        render() {
          return <TreeSelect placeholder={'please select address'} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':popupProps', () => {
      const popupProps = {
        overlayStyle: {
          width: '500px',
        },
      };
      const wrapper = mount({
        render() {
          return <TreeSelect popupProps={popupProps} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':size', () => {
      const wrapper = mount({
        render() {
          return <TreeSelect size={'large'} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':treeProps', () => {
      const treeProps = {
        keys: {
          label: 'name',
          value: 'pinyin',
          children: 'children',
        },
      };
      const wrapper = mount({
        render() {
          return <TreeSelect data={propsOptions} treeProps={treeProps} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':value', () => {
      const value = 'shenzhen';
      const wrapper = mount({
        render() {
          return <TreeSelect v-model={value} data={options} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':defaultValue', () => {
      const defaultValue = ['shenzhen', 'guangzhou'];
      const wrapper = mount({
        render() {
          return <TreeSelect defaultValue={defaultValue} data={options} multiple={true} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':valueType', () => {
      const wrapper = mount({
        render() {
          return <TreeSelect valueType={'object'} data={options} value={objectValue} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  // test events
  describe('@event', () => {
    it('remove', () => {
      const value = ['shenzhen'];
      const wrapper = mount({
        render() {
          return <TreeSelect data={options} value={value} multiple={true} />;
        },
      });
      const treeSelectWrapper = wrapper.findComponent(TreeSelect);
      const closeIconWrapper = wrapper.find('.t-icon-close');
      closeIconWrapper.trigger('click');
      expect(treeSelectWrapper.emitted().remove).toBeTruthy();
    });
    it('search', () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return <TreeSelect data={options} filterable={true} onSearch={fn} />;
        },
      });
      const treeSelectWrapper = wrapper.findComponent(TreeSelect);
      const inputElement = wrapper.find('input');
      inputElement.setValue('shenzhen');
      expect(treeSelectWrapper.emitted().search).toBeTruthy();
    });
  });

  // test slots
  describe('<slot>', () => {
    it('<collapsedItems>', () => {
      const wrapper = mount({
        render() {
          return (
            <TreeSelect>
              <div slot="collapsedItems">
                <span>更多...</span>
              </div>
            </TreeSelect>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it('<empty>', () => {
      const wrapper = mount({
        render() {
          return (
            <TreeSelect>
              <div slot="empty">no data</div>
            </TreeSelect>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it('<loadingText>', () => {
      const wrapper = mount({
        render() {
          return (
            <TreeSelect loading={true}>
              <div slot="loadingText">data loading</div>
            </TreeSelect>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it('<prefixIcon>', () => {
      const wrapper = mount({
        render() {
          return (
            <TreeSelect>
              <div slot="prefixIcon">
                <i>icon</i>
              </div>
            </TreeSelect>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  // test exposure function
  describe('function', () => {
    it(':collapsedItems', () => {
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      const renderCollapsedItems = (h) => (
        <div>
          <span>更多...</span>
        </div>
      );
      const wrapper = mount({
        render() {
          return <TreeSelect collapsedItems={renderCollapsedItems} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':empty', () => {
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      const renderEmpty = (h) => <div>no data</div>;
      const wrapper = mount({
        render() {
          return <TreeSelect empty={renderEmpty}></TreeSelect>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':loadingText', () => {
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      const renderLoadingText = (h) => <div>data loading</div>;
      const wrapper = mount({
        render() {
          return <TreeSelect loading={true} loadingText={renderLoadingText} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':prefixIcon', () => {
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      const renderPrefixIcon = (h) => (
        <div>
          <i>icon</i>
        </div>
      );
      const wrapper = mount({
        render() {
          return <TreeSelect prefixIcon={renderPrefixIcon} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
