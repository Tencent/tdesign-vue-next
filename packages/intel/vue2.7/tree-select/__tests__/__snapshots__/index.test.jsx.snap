// Vitest Snapshot v1

exports[`TreeSelect > :props > :clearable 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > :props > :defaultValue 1`] = `
<div
  class="t-select-input t-select-input--multiple t-tree-select"
>
  <div
    class="t-input__wrap t-tag-input t-tag-input--break-line t-tag-input--with-tag t-tag-input__with-suffix-icon"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      >
        <div
          class="t-tag t-tag--default t-tag--dark t-tag--ellipsis t-tag--close"
        >
          <span
            class="t-tag--text"
            style="max-width: 300px;"
          >
            深圳市
          </span>
          <svg
            class="t-icon t-icon-close t-tag__icon-close"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M7.05 5.64L12 10.59l4.95-4.95 1.41 1.41L13.41 12l4.95 4.95-1.41 1.41L12 13.41l-4.95 4.95-1.41-1.41L10.59 12 5.64 7.05l1.41-1.41z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="t-tag t-tag--default t-tag--dark t-tag--ellipsis t-tag--close"
        >
          <span
            class="t-tag--text"
            style="max-width: 300px;"
          >
            广州市
          </span>
          <svg
            class="t-icon t-icon-close t-tag__icon-close"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M7.05 5.64L12 10.59l4.95-4.95 1.41 1.41L13.41 12l4.95 4.95-1.41 1.41L12 13.41l-4.95 4.95-1.41-1.41L10.59 12 5.64 7.05l1.41-1.41z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <input
        autocomplete=""
        class="t-input__inner t-input--soft-hidden"
        placeholder=""
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__input-pre"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > :props > :disabled 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-disabled t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        disabled="disabled"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow t-fake-arrow--disable"
          disabled="disabled"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > :props > :empty string 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > :props > :filterable 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        type="text"
        unselectable="off"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > :props > :loading 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <div
          class="t-loading--center t-size-s t-loading"
        >
          <svg
            class="t-loading__gradient t-icon-loading"
            height="1em"
            size="small"
            version="1.1"
            viewBox="0 0 12 12"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <foreignobject
              height="12"
              width="12"
              x="0"
              y="0"
            >
              <div
                class="t-loading__gradient-conic"
              />
            </foreignobject>
          </svg>
        </div>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > :props > :loadingText 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > :props > :max 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > :props > :minCollapsedNum 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > :props > :multiple 1`] = `
<div
  class="t-select-input t-select-input--multiple t-tree-select"
>
  <div
    class="t-input__wrap t-tag-input t-tag-input--break-line t-tag-input--with-tag t-tag-input__with-suffix-icon"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      >
        <div
          class="t-tag t-tag--default t-tag--dark t-tag--ellipsis t-tag--close"
        >
          <span
            class="t-tag--text"
            style="max-width: 300px;"
          />
          <svg
            class="t-icon t-icon-close t-tag__icon-close"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M7.05 5.64L12 10.59l4.95-4.95 1.41 1.41L13.41 12l4.95 4.95-1.41 1.41L12 13.41l-4.95 4.95-1.41-1.41L10.59 12 5.64 7.05l1.41-1.41z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <input
        autocomplete=""
        class="t-input__inner t-input--soft-hidden"
        placeholder=""
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__input-pre"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > :props > :placeholder 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="please select address"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > :props > :popupProps 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > :props > :size 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-size-l t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > :props > :treeProps 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > :props > :value 1`] = `
<div
  class="t-select-input t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > :props > :valueType 1`] = `
<div
  class="t-select-input t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > <slot> > <collapsedItems> 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > <slot> > <empty> 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > <slot> > <loadingText> 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <div
          class="t-loading--center t-size-s t-loading"
        >
          <svg
            class="t-loading__gradient t-icon-loading"
            height="1em"
            size="small"
            version="1.1"
            viewBox="0 0 12 12"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <foreignobject
              height="12"
              width="12"
              x="0"
              y="0"
            >
              <div
                class="t-loading__gradient-conic"
              />
            </foreignobject>
          </svg>
        </div>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > <slot> > <prefixIcon> 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      >
        <div>
          <i>
            icon
          </i>
        </div>
      </div>
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > function > :collapsedItems 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > function > :empty 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > function > :loadingText 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      />
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <div
          class="t-loading--center t-size-s t-loading"
        >
          <svg
            class="t-loading__gradient t-icon-loading"
            height="1em"
            size="small"
            version="1.1"
            viewBox="0 0 12 12"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <foreignobject
              height="12"
              width="12"
              x="0"
              y="0"
            >
              <div
                class="t-loading__gradient-conic"
              />
            </foreignobject>
          </svg>
        </div>
      </span>
    </div>
  </div>
</div>
`;

exports[`TreeSelect > function > :prefixIcon 1`] = `
<div
  class="t-select-input t-select-input--empty t-tree-select"
>
  <div
    class="t-input__wrap"
  >
    <div
      class="t-input t-is-readonly t-input--prefix t-input--suffix"
    >
      <div
        class="t-input__prefix"
      >
        <div>
          <i>
            icon
          </i>
        </div>
      </div>
      <input
        autocomplete=""
        class="t-input__inner"
        placeholder="请选择"
        readonly="readonly"
        type="text"
        unselectable="on"
      />
      <span
        class="t-input__suffix t-input__suffix-icon"
      >
        <svg
          class="t-fake-arrow"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
